### ISSUE TITLE
Incorrect nullability of `ref` variables

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#33074]
Warnings are reported for each dereference:
```C#
#pragma warning disable 8618

class C
{
    internal object? F;
}

class Program
{
    static void F1(ref object? x, ref int? y, ref C z)
    {
        x = 1;
        x.ToString();

        y = 2;
        _ = y.Value;

        z = new C() { F = 3 };
        z.F.ToString();
    }
}
```
Result:
```
(13,9): warning CS8602: Possible dereference of a null reference.
(16,13): warning CS8629: Nullable value type may be null.
(19,9): warning CS8602: Possible dereference of a null reference.
```

**Comment by cston** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/25868.
